compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -386,7 +386,40 @@ class OMR_EXTENSIBLE CodeGenerator
 
    TR_HasRandomGenerator randomizer;
 
-   bool supportsAtomicAdd() {return false;}
+   /** \brief
+    *     Determines whether intrinsics on the following atomic symbols is supported:
+    *
+    *     - atomicAdd32BitSymbol
+    *     - atomicAdd64BitSymbol
+    *     - atomicFetchAndAdd32BitSymbol
+    *     - atomicFetchAndAdd64BitSymbol
+    *
+    *  \return
+    *     \c true if intrinsics are supported; \c false otherwise.
+    */
+   bool supportsAtomicAdd()
+      {
+      return false;
+      }
+
+   /** \brief
+    *     Determines whether intrinsics on the following atomic symbols is supported:
+    *
+    *     - atomicSwap32BitSymbol
+    *     - atomicSwap64BitSymbol
+    *     - atomicCompareAndSwap32BitSymbol
+    *     - atomicCompareAndSwap64BitSymbol
+    *     - atomicCompareAndSwapReturnValue32BitSymbol
+    *     - atomicCompareAndSwapReturnValue64BitSymbol
+    *
+    *  \return
+    *     \c true if intrinsics are supported; \c false otherwise.
+    */
+   bool supportsAtomicSwap()
```

Comment:
I added this query because on Z we can use compare and swap on all supported architecture levels where as atomic addition is only available on z196 onwards. We also need to be using the existing `supportsAtomicAdd` query in downstream projects.

Reply:
Fixed in a7704d1